src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copy = { ...state };
+  const res = [];
+
+  for (const action of actions) {
+    switch (true) {
+      case action.type === 'addProperties':
+        Object.assign(copy, action.extraData);
+        res.push({ ...copy });
+        break;
+
+      case action.type === 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          if (copy.hasOwnProperty(keyToRemove)) {
+            delete copy[keyToRemove];
+          }
+        }
+
+        res.push({ ...copy });
```

Comment:
you could push just after switch, not in every case

Reply:
> you could push just after switch, not in every case

ty, done